### PR TITLE
Updated Hashes

### PR DESCRIPTION
**Pull request by JohnnyCrazy**
Updated Hashes and refactored old hashes.

@crosire do you really want to remove all `_0x`-entries? I could modify the native-generator to do that, but I wanted to be sure before I do

### CODE REVIEW

**Comment by crosire** [comment]
In 3.0, yes. The enum is a massive beast. Any way to reduce its size is welcomed :smile:.

**Comment by JohnnyCrazy** [comment]
Alright, will update it :+1:

**Comment by JohnnyCrazy** [comment]
Done, I also published the generator @ [JohnnyCrazy/scripthookvdotnet-nativegen](https://github.com/JohnnyCrazy/scripthookvdotnet-nativegen)

You were right, this removes a lot of lines.

Ready for 3.0 :+1:

**Comment by zorg93** [comment]
When you say remove all _0x natives do you mean all of them or just the ones that have actual readable names as well

**Comment by zorg93** [comment]
Removing all _0x natives if they don't have an actual name means people would have to cast the raw native hashes for undefined natives which they could easily make a mistake doing causing script hook v to throw an unknown native error

**Comment by JohnnyCrazy** [comment]
If a native has a readable name, then there is no _0x name for it? _0x is only used for natives without names

**Comment by crosire** [comment]
It was used for all natives. Even those that have a name (for backwards compatibility). But they shouldn't be there either way. If people make a mistake, the error pop-up should be clear enough.

File: source/scripting/Scaleform.cpp [hunk 1]
```diff
@@ -97,10 +97,10 @@ namespace GTA
 	}
 	void Scaleform::Render3D(Math::Vector3 position, Math::Vector3 rotation, Math::Vector3 scale)
 	{
-		Native::Function::Call(Native::Hash::_0x1CE592FDC749D6F5, Handle, position.X, position.Y, position.Z, rotation.X, rotation.Y, rotation.Z, 2.0f, 2.0f, 1.0f, scale.X, scale.Y, scale.Z, 2);
+		Native::Function::Call(Native::Hash::_DRAW_SCALEFORM_MOVIE_3D_NON_ADDITIVE, Handle, position.X, position.Y, position.Z, rotation.X, rotation.Y, rotation.Z, 2.0f, 2.0f, 1.0f, scale.X, scale.Y, scale.Z, 2);
 	}
 	void Scaleform::Render3DAdditive(Math::Vector3 position, Math::Vector3 rotation, Math::Vector3 scale)
 	{
-		Native::Function::Call(Native::Hash::_0x87D51D72255D4E78, Handle, position.X, position.Y, position.Z, rotation.X, rotation.Y, rotation.Z, 2.0f, 2.0f, 1.0f, scale.X, scale.Y, scale.Z, 2);
+		Native::Function::Call(Native::Hash::_DRAW_SCALEFORM_MOVIE_3D_NON_ADDITIVE, Handle, position.X, position.Y, position.Z, rotation.X, rotation.Y, rotation.Z, 2.0f, 2.0f, 1.0f, scale.X, scale.Y, scale.Z, 2);
```
Comment (Fireboyd78) [inline source/scripting/Scaleform.cpp]: Wrong natives used

Comment (Fireboyd78) [inline source/scripting/Scaleform.cpp]: Two different natives but you duped the same one, make sure you fix this.

**Comment by Fireboyd78** [comment]
Sorry for the double line comment, I dunno why I can't delete the other one (I'm on mobile).

**Comment by JohnnyCrazy** [comment]
@Fireboyd78 Ye, my bad, fixed

**Comment by Fireboyd78** [comment]
@JohnnyCrazy I just thought of something. There's quite a few native hashes that are the result of hash collisions. I don't know of any off hand, but maybe you should implement a sort of "whitelist" for confirmed natives (similar to Glokon's native hasher), and since unknown hashes start with "_", you could create a ""whitelist_verify.txt" file that gets spit out when you generate hashes.

**Comment by JohnnyCrazy** [comment]
@Fireboyd78  
I still don't quite get your point. What is an example for a native collision?

**Comment by Fireboyd78** [comment]
@JohnnyCrazy Tl;dr -- make a list of verified natives (a "whitelist") so when we generate hashes, it doesn't include hash collisions.

When a new native is detected, have it spit out that native to "whitelist_verify.txt" for further confirmation. If it seems to be correct, then add it to the whitelist and re-generate hashses.

Make sense now?

**Comment by crosire** [comment]
We don't generate the hashes from game data. They are downloaded from nativedb. So I don't understand your point either. We don't need to verify them? They already are.

**Comment by zorg93** [comment]
Natives from NativeDB are community driven and were past the point where people try to troll it. Most natives on there people have already detected obvious hash collisions

**Comment by Fireboyd78** [comment]
@crosire @zorg93 I have personally found many natives myself. I use NativeDB extensively. I've seen false natives that have taken months to be reverted, since Alexander Blade does not have any admins/moderators to remove "spam". People have to inform him of collisions, which get removed whenever he decides to get around to it.

**Comment by crosire** [comment]
Since nativedb is pretty much *the* documentation available, it is convenient if the names in ScriptHookVDotNet match. Even if they aren't always correct. In my opinion.

**Comment by JohnnyCrazy** [comment]
Merged into `dev_v3` in 6d42223935f90a6a0dea14e5f83237333fadc0ae

**Pull request closed without merging**